### PR TITLE
Allow User to Edit Expense Even if Draft Key was Passed in URL

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4902

Front End PR: https://github.com/opencollective/opencollective-frontend/pull/7081

### CODE REVIEW

**Comment by SudharakaP** [comment]
Correct me if I am wrong, but I don't see the necessity for this condition. The condition is only triggered when a `draftKey` is passed. If a `draftKey` is not passed the user can edit the expense. Ideally I think we should give the user the ability to edit he expense even if the draft key is passed or not as the user might be accessing the (possibly already submitted and pending) expense using the invite link. 🤔

**Review by znarf — CHANGES_REQUESTED** [review]
I don't think we should enter the `if (args.draftKey)` condition if the expense is not a DRAFT anymore.

**Comment by SudharakaP** [comment]
> I don't think we should enter the `if (args.draftKey)` condition if the expense is not a DRAFT anymore.

@znarf : Makes sense and feel that's a better solution. I've corrected the code to add that condition. 🤔

File: server/graphql/v2/input/ExpenseUpdateInput.ts [hunk 1]
```diff
@@ -67,5 +67,9 @@ export const ExpenseUpdateInput = new GraphQLInputObjectType({
       type: LocationInput,
       description: 'The address of the payee',
     },
+    status: {
+      type: GraphQLString,
+      description: 'The status of the expense such as DRAFT, PENDING etc',
+    },
```
Comment (znarf) [inline server/graphql/v2/input/ExpenseUpdateInput.ts:73]: Why is this needed? The type look wrong but instead of fixing it I'm wondering if this can be removed.

Comment (SudharakaP) [reply server/graphql/v2/input/ExpenseUpdateInput.ts:73]: This is needed because now we are passing the status as an arg for the `ExpenseUpdateInput` object. Specifically at, 

https://github.com/opencollective/opencollective-frontend/blob/db5ddcfcc0fd0bee15af15d73268116562a509b8/components/expenses/ExpenseForm.js#L117

We can probably remove this if we calculate the `existingExpense` outside of the `if` condition; 

https://github.com/opencollective/opencollective-api/blob/b9cdeb9b9dcc438a4d508769cc76a0a4934110e6/server/graphql/v2/mutation/ExpenseMutations.ts#L138-L143

However I didn't do this as passing it from the front-end seems straightforward and doesn't make an unnecessary db call. Do you have a better alternative? 🤔

Comment (znarf) [reply server/graphql/v2/input/ExpenseUpdateInput.ts:73]: Can we simply not pass it? 

@Betree what do you think?

Comment (Betree) [reply server/graphql/v2/input/ExpenseUpdateInput.ts:73]: I don't get why we'd need to pass a `status`, but it definitely looks wrong:
- What does it mean if someone passes `HELLO` as a status? (the current field is not an enum)
- What does it mean if someone passes `PAID` as a status?

`status` is not something we want people to be able to edit, it's an indication of the result from other mutations (`processExpense`). We need to find a better API there.

Comment (Betree) [reply server/graphql/v2/input/ExpenseUpdateInput.ts:73]: Can't we just omit `draftKey` if it's not a draft?

Comment (SudharakaP) [reply server/graphql/v2/input/ExpenseUpdateInput.ts]: That makes sense. I've refactored the front-end pr to omit the `draftKey` if it's not a draft. Thanks for the idea; that seems such a simple solution. 😉

**Comment by SudharakaP** [comment]
@znarf : I am closing this PR as this is no longer required; only the front-end PR is required as per the above comment; https://github.com/opencollective/opencollective-api/pull/6779#discussion_r743803180
